packages/common/src/ui/components/modals/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
+import { Grid, Typography } from '@mui/material';
+import { BasicModal } from '../BasicModal';
+import { AlertIcon } from '@common/icons';
+import { DialogButton, LoadingButton } from '../../buttons';
+
+interface ConfirmationModalProps {
+  open: boolean;
+  width?: number;
+  height?: number;
+  onConfirm: (() => void) | (() => Promise<void>);
+  onCancel: () => void;
+  title: string;
+  message: string;
+  Icon?: React.ReactNode;
+}
+
+export const ConfirmationModal = ({
+  open,
+  width = 400,
+  height = 200,
+  onConfirm,
+  title,
+  message,
+  onCancel,
+  Icon = <AlertIcon color="primary" />,
```

Comment:
wonder if the icon should be a prop; or an enum for alert, info, error 🤔

Reply:
that's a good idea!